keps/sig-node/20200310-ensure-secret-pulled-images.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Ensure Secret Pulled Images
+authors:
+  - "@mikebrow"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@Random-Liu"
+  - "@yujuhong"
+approvers:
+  - "@dchen1107"
+editor: N/A
+creation-date: 2020-03-10
+last-updated: 2020-03-10
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Ensure Secret Pulled Images
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Examples](#examples)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
```

Comment:
Should this be something like a `ImageCacheSharePolicy: pod, namespace, all`?
* pod: The image is only usable to this pod;
* namespace: The image is only usable to pods in the same namespace;
* all: The image is usable to all pods.

And this only affects images that need secret to pull, because public images are inherently sharable because they are public.

Reply:
Yes, this is idea I was talking about with you at the last kubecon, that would be better in the long run, passing the image cache policy over CRI to have the image service in the container runtime manage the image cache part of the contract. But you said at the time we should first do a small change just fixing what we already have in kubelet. Once we move to a image cache policy pull policies are no longer useful for the purpose of isolation.

Reply:
the difficulty is in solving it from the current pull policy model we have.. vs introducing a new type of policy and explaining how that conflicts or does not conflict with the pull policy.. and well who will manage the cache policy kublet or the runtimes.

Reply:
let's do a phone call and review :)